src/framework/mpas_timekeeping.F

```diff
@@ -486,20 +486,24 @@ type (MPAS_Time_type) function mpas_get_clock_time(clock, whichTime, ierr)
    end function mpas_get_clock_time
 
 
-   subroutine mpas_add_clock_alarm(clock, alarmID, alarmTime, alarmTimeInterval, ierr)
-! TODO: possibly add a stop time for recurring alarms
+   subroutine mpas_add_clock_alarm(clock, alarmID, alarmTime, alarmTimeInterval, &
+         alarmStartTime, alarmStopTime, ierr)
 
       implicit none
 
       type (MPAS_Clock_type), intent(inout) :: clock
       character (len=*), intent(in) :: alarmID
       type (MPAS_Time_type), intent(in) :: alarmTime
       type (MPAS_TimeInterval_type), intent(in), optional :: alarmTimeInterval
-      integer, intent(out), optional :: ierr
+      type (MPAS_Time_type), intent(in), optional :: alarmStartTime
+      type (MPAS_Time_type), intent(in), optional :: alarmStopTime
+      integer, intent(out),  optional :: ierr
 
       type (MPAS_Alarm_type), pointer :: alarmPtr
       integer :: threadNum
 
+      if (present(ierr)) ierr = ESMF_SUCCESS
```

Comment:
Since existing code compares the value of `ierr` returned by this subroutine with 0 to determine success, I'd suggest we set `ierr` to 0 rather than to `ESMF_SUCCESS`.